core/src/main/java/hudson/model/queue/Executables.java

```diff
@@ -71,5 +73,33 @@ public static long getEstimatedDurationFor(Executable e) {
             return e.getParent().getEstimatedDuration();
         }
     }
+    
+    /**
+     * Finds the executor running a given process.
+     * @param executable a possibly running executable
+     * @return the executor (possibly a {@link OneOffExecutor}) whose {@link Executor#getCurrentExecutable} matches that, or null
+     * @since TODO
+     */
+    public static @CheckForNull Executor getExecutor(Executable executable) {
```

Comment:
Feels like this belongs more to the `Executor` class, like `Executor.of(executable)`